### PR TITLE
Use SVG spritesheet for our icons

### PR DESCRIPTION
**Pull request by eric-gade**
## What does this PR do? 🛠️
**NOTE: This PR relates to #531 and closes that PR without merging in its changes**  

We have decided to follow the [pattern recommended by USWDS](https://designsystem.digital.gov/components/icon/#accessibility-icon), using SVG partial symbols and selectors to display icons. This gives us more control over the styling and accessibility qualities whenever we need.
  
This PR implements the following:
- A script that will take all icon SVG files, modify them slightly, and generate a single `spritesheet.svg` file.
- Updates hourly and daily blocks to return the plain filename (without extension) of each icon
- Updates hourly and daily block templates to use the `<use xlink:href>` style selection of a single icon from the spritesheet, along with necessary styling to make them appear as they already were

## What does the reviewer need to know? 🤔
To run the spritesheet compiler:
```shell
npm run compile-svg
```

### CODE REVIEW

**Comment by greg-does-weather** [comment]
This will close #521 too.

File: scripts/compile-svg-sprite.js [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+ * @return string A string with extraneous
+ * newlines removed
+ */
+const cleanupWhitespace = (str) => {
+  const lines = str.split("\n");
+  return lines.map(line => {
+    return line.replace(/^\w+\n$/, "");
+  }).filter(line => {
+    return line !== "";
+  }).join("\n");
```
Comment (greg-does-weather) [inline scripts/compile-svg-sprite.js:56]: I believe this multiline regex should work as well:

```suggestion
return str.replace(/\n+/gm, '\n');
```

File: scripts/compile-svg-sprite.js [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+ * Removes the XML header from the string
+ *
+ * @param str string The source xml string
+ * @return string The cleaned output xml string
+ */
+const removeXMLDeclaration = str => {
+  const lines = str.split("\n");
+  return lines.filter(line => {
+    return !line.startsWith("<?xml");
+  }).join("\n");
```
Comment (greg-does-weather) [inline scripts/compile-svg-sprite.js:69]: I'm partial to regular expressions, but I'll defer to you on readability. I think what you have here is probably more readable, tbh.

```suggestion
 return str.replace(/\n<?xml[^\n]*/gm, "");
```

**Review by greg-does-weather — COMMENTED** [review]
This is clever. I like it a lot! But you're going to hate me.

When I run it locally, I can't actually see the icons because they have a white fill. And... it turns out... you can't target stroke and fill on SVG sprites with CSS. 😬

Instead of setting the fill color, we *could* apply a brightness filter:

```css
.icon svg {
  filter: brightness(0);
}
```

That starts to fill icky, though. Or at least less obvious than `fill: black;`. I'm sorry, I feel like I sent you down this path and now it's a bad path.

File: scripts/compile-svg-sprite.js (hunk 1 above)
Comment (eric-gade) [reply scripts/compile-svg-sprite.js:56]: I'll mess with the regex. There are places where the SVGs were generated (cough cough, hello Illustrator) with lines that have only spaces and/or tabs

Comment (greg-does-weather) [reply scripts/compile-svg-sprite.js:56]: Oooh, I missed that nuance.

**Comment by partly-igor** [comment]
> When I run it locally, I can't actually see the icons because they have a white fill. And... it turns out... you can't target stroke and fill on SVG sprites with CSS. 😬

Hopefully I am not overstepping here, and I might be missing something, but it seems like it's possible in general to target with CSS

image

**Comment by mgwalker** [comment]
You're definitely not overstepping, and you're also correct. I think the problem I was seeing was elsewhere and it resulted in the style I was setting just not being applied. But my "not possible" statement was wrong.

**Comment by eric-gade** [comment]
@partly-igor to bring you up to speed: it looks like they don't accept the styling in FF. There's all kind of chatter online about what to do but none of it seems to work when I try it...

**Comment by eric-gade** [comment]
Alright @greg-does-weather cracked the case! Because we are still loading individual SVGs elsethere on the page, and these SVGs have style tags that define the `st0` class, that style is available globally. Our spritesheet was removing style tags but not class attributes that referenced `st0` (and the devtools were not forthcoming about where this was happening).
  
The solution, pushed in https://github.com/weather-gov/weather.gov/pull/543/commits/5ff33c324b04efdb85cfcdcc74fba74d022e4017 , was to simply remove all class attributes from any element in the spritesheet.

**Review by greg-does-weather — APPROVED** [review]

**Pull request merged**